connman/vpn/plugins/openvpn.c

```diff
@@ -357,38 +366,52 @@ static int ov_save(struct vpn_provider *provider, GKeyFile *keyfile)
 static int task_append_config_data(struct vpn_provider *provider,
 					struct connman_task *task)
 {
-	const char *option;
 	bool block_ipv6 = false;
 	int i;
 
 	for (i = 0; i < (int)ARRAY_SIZE(ov_options); i++) {
-		if (!ov_options[i].ov_opt)
-			continue;
+		const char *ov_opt = ov_options[i].ov_opt;
+		const char *cm_opt = ov_options[i].cm_opt;
+		const char *option = NULL;
 
-		/*
-		 * In case the option is without value, i.e. a boolean toggle
-		 * then ignore the value when option is disabled.
-		 */
-		if (!ov_options[i].has_value && !vpn_provider_get_boolean(
-					provider, ov_options[i].cm_opt, false))
+		switch (ov_options[i].opt_type) {
+		case OPT_NONE:
 			continue;
 
-		option = vpn_provider_get_string(provider,
-					ov_options[i].cm_opt);
-		if (!option)
-			continue;
+		case OPT_STRING:
+			if (!ov_opt)
```

Comment:
This seems a bit redundant now with the addition of the OPT_NONE. No real harm with keeping it though.

Reply:
I was thinking whether to use it or not. Especially since `connman_task_add_argument()` does check that name exists and returns error if it does not -> `task_append_config_data()` would return in that case.. Maybe this could be left to avoid misuse in `ov_options[]`.